src/Operation/Factory/SubresourceOperationFactory.php

```diff
@@ -79,6 +81,15 @@ private function computeSubresourceOperations(string $resourceClass, array &$tre
             $subresourceMetadata = $this->resourceMetadataFactory->create($subresourceClass);
 
             $visiting = "$resourceClass $property $subresourceClass";
+
+            // Handle maxDepth
+            if ($rootResourceClass === $resourceClass) {
+                $maxDepth = $subresource->getMaxDepth();
+            }
+
+            if(null !== $maxDepth && count($visited) >= $maxDepth) {
```

Comment:
use the `$depth` counter instead of count or there is no point :p.

Reply:
yes my bad... I commit/push too quickly